src/App.jsx

```diff
@@ -14,45 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ALPHABET = 'alphabetically';
+
+function getPreparedGoods(goods, sorted, reversed) {
+  const preparedGoods = [...goodsFromServer];
+
+  if (!sorted && reversed) {
+    return preparedGoods.reverse();
+  }
+
+  if (sorted) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sorted) {
+        case SORT_BY_ALPHABET:
+          return reversed
+            ? good2.localeCompare(good1)
+            : good1.localeCompare(good2);
+
+        case SORT_BY_LENGTH:
+          return reversed
+            ? good2.length - good1.length
+            : good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setRevers] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/96409563/1dbead00-2c9d-49a9-bb51-65c3e124c084)